mimic/model/heat_objects.py

```diff
@@ -149,6 +151,38 @@ def default_create_behavior(collection, request, body, absolutize_url):
     return json.dumps(response)
 
 
+@stack_update.declare_default_behavior
+def default_update_behavior(collection, request, stack_name, stack_id):
+    """
+    Successfully update a stack as long as it exists. Updates the stacks status.
+    """
+    stack = collection.stack_by_id(stack_id)
```

Comment:
Non-blocking suggestion: Code to check stack based on id and returning 404 can be in a decorator.
```python
def stack_exists(f):
    def decorator(collection, request, stack_name, stack_id):
        stack = collection.stack_by_id(stack_id):
        if not stack:
            request.setResponseCode(404); return b''
        return f(collection, request, stack)
    return decorator

@stack_exists
@stack_update.declare_default_behavior
def default_update_behavior(collection, request, stack):
...
```
This is merely a suggestion. Feel free to ignore :)